### PR TITLE
Support Precompiled Kernels in Orochi

### PR DESCRIPTION
**Pull request by KaoCC**
Support Precompiled Bitcode in Orochi

### CODE REVIEW

File: Orochi/OrochiUtils.h [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+	if( !( x ) )                                                                                                                                                                                                                                               \
+	{                                                                                                                                                                                                                                                          \
+		__debugbreak();                                                                                                                                                                                                                                        \
+	}
 #else
-	#define OROASSERT(x, y) if(!(x)) {;}
+#define OROASSERT( x, y )                                                                                                                                                                                                                                      \
```
Comment (meistdan) [inline Orochi/OrochiUtils.h]: Suggestion: On Linux, you can use th following construct:
#elif defined(__GNUC__)
#include <signal.h>
#define OROASSERT(cond) if (!(cond)) { raise(SIGTRAP); }
#else
#define OROASSERT(cond) 
#endif

File: ParallelPrimitives/RadixSort.cpp [hunk 2]
```diff
@@ -5,13 +5,17 @@
 #include <numeric>
 
 #if defined( ORO_PP_LOAD_FROM_STRING )
-#include <ParallelPrimitives/cache/Kernels.h>
 #include <ParallelPrimitives/cache/KernelArgs.h>
+#include <ParallelPrimitives/cache/Kernels.h>
```
Comment (meistdan) [inline ParallelPrimitives/RadixSort.cpp:11]: Did you use the automatic include sorting? If I remember correctly, I had some issue with baking when I used the automatic include sorting. I think it must be in this order:
#include <ParallelPrimitives/cache/Kernels.h>
#include <ParallelPrimitives/cache/KernelArgs.h>

**Review by meistdan — CHANGES_REQUESTED** [review]
I'm not sure about the bitcode linking part, so I'll leave it to others who know more about that.

Otherwise, it looks good, just check whether the include order doesn't break the baking path.

File: ParallelPrimitives/RadixSort.cpp (hunk 2 above)
Comment (KaoCC) [reply ParallelPrimitives/RadixSort.cpp:11]: I'll add 
```
// clang-format off
...
// clang-format on
```

to turn off formatting here

File: Orochi/OrochiUtils.h (hunk 1 above)
Comment (KaoCC) [reply Orochi/OrochiUtils.h]: I rewrote the entire function

**Review by meistdan — APPROVED** [review]
Looks good :-)
Thank you

**Comment by jammm** [comment]
Other than the description and what we talked about internally re. another project, it looks good to me.

**Review by takahiroharada — APPROVED** [review]

**Pull request merged**